app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/stackTrace/stackTrace.jsx

```diff
@@ -0,0 +1,184 @@
+    this.state = {
+      stackTraceItem: null,
+      isStackTraceItemOnThisPage: false,
+      activePage: props.pagination.number,
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchStackTrace();
+  }
+
+  static getDerivedStateFromProps(props, state) {
+    return state.activePage !== props.pagination.number
+      ? {
+          activePage: props.pagination.number,
+          isStackTraceItemOnThisPage: StackTrace.checkIfStackTraceItemOnThisPage(
+            props.logItems,
+            state.stackTraceItem,
+          ),
+        }
+      : null;
+  }
+
+  fetchStackTrace = () => {
+    const { logItems, projectId, logItemId } = this.props;
+    this.setState({
+      loading: true,
+    });
+    fetch(URLS.logItemStackTrace(projectId, logItemId))
+      .then(({ content }) => {
+        const stackTraceItem = (content.length && content[0]) || null;
+        this.setState({
+          stackTraceItem,
+          isStackTraceItemOnThisPage: StackTrace.checkIfStackTraceItemOnThisPage(
+            logItems,
+            stackTraceItem,
+          ),
+          loading: false,
+        });
+      })
+      .catch(() => {
+        this.setState({
+          loading: false,
+        });
+      });
+  };
+
+  goToMessageInLog = ({ wasClicked, targetPageNumber }) => {
+    const { isStackTraceItemOnThisPage, stackTraceItem, activePage } = this.state;
+
+    if (isStackTraceItemOnThisPage) {
+      const targetItem = document.getElementById(stackTraceItem.id);
+      targetItem.scrollIntoView({ behavior: 'smooth' });
+      let highlightElement = document.getElementById('highlight');
+      if (!highlightElement) {
+        highlightElement = document.createElement('div');
+        highlightElement.setAttribute('id', 'highlight');
+        highlightElement.classList.add(cx('highlight'));
+        targetItem.appendChild(highlightElement);
+        setTimeout(() => targetItem.classList.add(cx('hide-highlight')), 0);
+      }
+      return;
+    }
+    if (activePage === targetPageNumber) {
+      return;
+    }
+    if (wasClicked) {
+      setTimeout(() => this.goToMessageInLog({ wasClicked: true }), 400);
+      return;
+    }
+    this.fetchMessageLocation();
+  };
+
+  fetchMessageLocation = () => {
+    const { projectId, logItemId, pagination, pageProperties, onChangePage } = this.props;
+    const { stackTraceItem } = this.state;
+    const filterLevel = pageProperties['filter.gte.level'] || DEFAULT_LOG_LEVEL;
+    return fetch(
+      URLS.logItemStackTraceMessageLocation(
+        projectId,
+        logItemId,
+        stackTraceItem.id,
+        pagination.size,
+        filterLevel,
+      ),
+    ).then(({ number }) => {
+      onChangePage(number);
+      this.goToMessageInLog({ wasClicked: true, targetPageNumber: number });
+    });
+  };
+
+  renderStackTraceMessage = () => {
+    const { intl } = this.props;
+    const { stackTraceItem } = this.state;
+
+    return (
+      // this.goToMessageInLog
```

Comment:
Please remove if don't needed.

Reply:
Can I comment it with `TODO` ?
These comments are necessary to show what we need to do for correct work of stack trace when  the list of logs will be available.